### PR TITLE
Updated server error handling in Requester to account for 5xx errors …

### PR DESCRIPTION
**Pull request by MinnDevelopment**
…rather than simply buttflare html responses

### CODE REVIEW

**Comment by Almighty-Alpaca** [comment]
wouldn't it be nice if you could set the number retry attempts yourself in `JDA`/`JDABuilder` ?

**Comment by DV8FromTheWorld** [comment]
Provide a way to customize the amount of attemps via JDABuilder. Default to 3 attempts (which is what it was before, this has made it 4).

**Comment by MinnDevelopment** [comment]
@DV8FromTheWorld what should the maximum amount of attempts be

**Comment by DV8FromTheWorld** [comment]
I guess like 10 i suppose. Perhaps it shouldn't be customizable. Is there really a reason to attempt a request more than 3 times if we get a 5xx error? it means something is wrong with discord..

**Comment by DV8FromTheWorld** [comment]
This is acceptable after the trial attempts is changed from `4` to `3`.

File: src/main/java/net/dv8tion/jda/core/requests/Requester.java [hunk 1]
```diff
@@ -115,33 +114,34 @@ public Long execute(Request apiRequest)
...
-                response = request.asString();
-                attempt++;
             }
-            if (response.getBody() != null && response.getBody().startsWith("<"))
+            while (attempt < 3 && response.getStatus() >= 500);
+
+            if (response.getStatus() >= 500)
             {
-                //Epic failure due to cloudflare. Attempted 4 times.
+                //Epic failure from other end. Attempted 4 times.
```
Comment (Almighty-Alpaca) [inline src/main/java/net/dv8tion/jda/core/requests/Requester.java]: `Attempted 4 times.` -> `Attempted 3 times.`

**Pull request merged**